clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -69,57 +70,74 @@ def run(self, progress_callback):
             append_datasets_to_list(recipe_outputs_dict, output_datasets)
 
         # Identify Flow input/outputs:
-        flow_inputs = [x for x in input_datasets if x not in output_datasets]
-        flow_outputs = [x for x in output_datasets if x not in input_datasets]
+        flow_inputs = [dataset for dataset in input_datasets if dataset not in output_datasets]
+        flow_outputs = [dataset for dataset in output_datasets if dataset not in input_datasets]
         logging.info("Found {} FLOW INPUT datasets: {}".format(str(len(flow_inputs)),
                                                                str(flow_inputs)))
         logging.info("Found {} FLOW OUTPUT datasets: {}".format(str(len(flow_outputs)),
                                                                 str(flow_outputs)))
 
+        # Identify Intermediate datasets:
+        intermediate_datasets = [dataset["name"] for dataset in all_datasets if dataset["name"] not in flow_inputs + flow_outputs]
+        logging.info("Found {} INTERMEDIATE datasets: {}".format(str(len(intermediate_datasets)),
+                                                                str(intermediate_datasets)))
+
         # Identify shared datasets:
-        shared_objs = project.get_settings().settings["exposedObjects"]["objects"]
-        shared_datasets = [x["localName"] for x in shared_objs if x["type"]=="DATASET"]
+        shared_objects = project.get_settings().settings["exposedObjects"]["objects"]
+        shared_datasets = [object["localName"] for object in shared_objects if object["type"]=="DATASET"]
         logging.info("Found {} SHARED datasets: {}".format(str(len(shared_datasets)),
                                                            str(shared_datasets)))
 
-        # Identify partitioned (partd) datasets:
-        is_partd = lambda x: len(x["partitioning"]["dimensions"]) > 0
-        partd_datasets = [x["name"] for x in all_datasets if is_partd(x)]
-        logging.info("Found {} PARTITIONED datasets: {}".format(str(len(partd_datasets)),
-                                                                str(partd_datasets)))
+        # Identify partitioned datasets:
+        is_partitioned = lambda x: len(dataset["partitioning"]["dimensions"]) > 0
+        partitioned_datasets = [dataset["name"] for dataset in all_datasets if is_partitioned(dataset)]
```

Comment:
the macro fails at line 93
```suggestion
        is_partitioned = lambda dataset: len(dataset["partitioning"]["dimensions"]) > 0
        partitioned_datasets = [dataset["name"] for dataset in all_datasets if is_partitioned(dataset)]
```
![Capture d’écran 2022-01-26 à 15 08 18](https://user-images.githubusercontent.com/77485161/151177841-dbce66c2-b6e8-4ee1-a806-dfb6d6927d5b.png)